### DISCUSSION TITLE
Multiple Span support between 2 nodes

### DISCUSSION THREAD

**Discussion by SaiBhyri** [Telecominfraproject/oopt-gnpy#484] [Q&A]
Hi Team,

Do we support multiple spans between 2 given nodes? 
For eg: Say we have 3 parallel spans (Span 1, Span 2, Span 3) between Site 1 and Site 2. 

If yes, how do we force Path and HW?
For Path, I find that only Site names can be given and not Span names.
For HW, I find that there is no Side concept and only Site concept.

Thanks,
Sai

**Comment by EstherLerouzic** [comment]
Dear Sai,
Parallel links are **partially** supported in JSON files and **not at all with excel input files**, and with amplified links only.
Path computation on one direction and signal propagation are correctly handled, but opposite direction of a path is not correctly computed, because OMS (optical multiplex section) are not correctly associated with their reversed direction OMS.
As a result, you can use the program to compute feasibility on parallel spans from the source to destination but not on the reversed direction of this path.
This is an identified missing feature: https://github.com/Telecominfraproject/oopt-gnpy/issues/425
Here is an example of 3 parallel links between two adjacent ROADMs, with a service example file to force path on one or the other link.
I have added the amplifier in the logs to show you that the path computation effectively selects three parallel links.
[services_parallel.json](https://github.com/Telecominfraproject/oopt-gnpy/files/13615428/services_parallel.json)
[topo_parallel.json](https://github.com/Telecominfraproject/oopt-gnpy/files/13615429/topo_parallel.json)


```
Computing all paths with constraints
Propagating on selected path
INFO:gnpy.topology.request:
	request 0
	Computing path from trx SITE1 to trx SITE2
	with path constraint: ['trx SITE1', 'trx SITE2']
	Computed path (roadms):['roadm SITE1', 'east edfa in SITE1 to SITE2', 'west edfa in SITE2 to SITE1', 'roadm SITE2']
INFO:gnpy.topology.request:
	request 1
	Computing path from trx SITE1 to trx SITE2
	with path constraint: ['trx SITE1', 'trx SITE2']
	Computed path (roadms):['roadm SITE1', 'Edfa_booster_roadm SITE1_to_fiber (SITE1 → SITE2) bis', 'Edfa_preamp_roadm SITE2_from_fiber (SITE1 → SITE2) bis', 'roadm SITE2']
INFO:gnpy.topology.request:
	request 3
	Computing path from trx SITE1 to trx SITE2
	with path constraint: ['trx SITE1', 'east edfa in SITE1 to ILA1', 'trx SITE2']
	Computed path (roadms):['roadm SITE1', 'east edfa in SITE1 to ILA1', 'east edfa in ILA1 to SITE2', 'west edfa in SITE2 to ILA1', 'roadm SITE2']
INFO:gnpy.topology.request:
	request 4
	Computing path from trx SITE1 to trx SITE2
	with path constraint: ['trx SITE1', 'east edfa in SITE1 to SITE2', 'trx SITE2']
	Computed path (roadms):['roadm SITE1', 'east edfa in SITE1 to SITE2', 'west edfa in SITE2 to SITE1', 'roadm SITE2']
Result summary
req id   demand                    GSNR@bandwidth A-Z (Z-A)   GSNR@0.1nm A-Z (Z-A)        Receiver minOSNR               mode                    Gbit/s              nb of tsp pairs      N,M or blocking reason 
0       trx SITE1 to trx SITE2 :            24.99                    29.07                      14                     mode 1                   100.0                      1                   ([-284],[4])      
1       trx SITE1 to trx SITE2 :            25.47                    29.55                      14                     mode 1                   100.0                      1                   ([-276],[4])      
3       trx SITE1 to trx SITE2 :            20.79                    24.87                      14                     mode 1                   100.0                      1                   ([-268],[4])      
4       trx SITE1 to trx SITE2 :            24.99                    29.07                      14                     mode 1                   100.0                      1                   ([-260],[4]) 
```
